### PR TITLE
4 - GET /api/articles/:article_id

### PR DESCRIPTION
**Pull request by Daniel-N98**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -33,9 +33,37 @@ describe("Status: 200, GET /api/topics", () => {
...
+      });
+  });
+  test("Status: 404, handles error when no article exists with the given ID", () => {
+    return request(app)
+      .get("/api/articles/15")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.message).toBe("Not found");
+      });
+  });
```
Comment (popatre) [inline __tests__/app.test.js:68]: Needs a test:

- [x] Status 400 - invalid id

**Review by popatre — CHANGES_REQUESTED** [review]
Just needs a bad request test and handler

File: __tests__/app.test.js (hunk 1 above)
Comment (xViolaine) [reply __tests__/app.test.js:68]: I can see that you amended this test, but you actually need to _add_ another test, not amend the original one.

You want to make sure you have two test cases:
- One for if the ID is valid, but it doesn't exist - like you had before, e.g. /api/articles/15
- One for if the ID is invalid - like if it was a string, e.g. /api/articles/InvalidString

**Review by xViolaine — CHANGES_REQUESTED** [review]
Hello,
Please add another test so both cases I outlined in my comment are handled! 😄

File: __tests__/app.test.js [hunk 2]
```diff
@@ -62,6 +62,14 @@ describe("GET /api/articles/:article_id", () => {
     return request(app)
       .get("/api/articles/15")
       .expect(400)
+      .then(({ body }) => {
```
Comment (popatre) [inline __tests__/app.test.js:65]: These need switching around 404 - article not found. 
You've asked for no 15 ( a valid number)

15 was not found 404

File: __tests__/app.test.js [hunk 3]
```diff
@@ -62,6 +62,14 @@ describe("GET /api/articles/:article_id", () => {
...
+        expect(body.message).toBe("Not found");
+      });
+  });
+  test("Status: 404, handles error when Id is invalid", () => {
```
Comment (popatre) [inline __tests__/app.test.js]: You've asked for /notAnId

This is a not a valid id (its not a number)

This is a bad request - 400

**Review by popatre — CHANGES_REQUESTED** [review]
You need to switch them around

**Pull request merged**